_test/test_sqw_class/test_equal_to_tol.m

```diff
@@ -41,7 +45,61 @@ function delete(obj)
 
     function test_the_same_sqw_objects_are_equal(obj)
         sqw_copy = obj.sqw_2d;
-        assertTrue(equal_to_tol(obj.sqw_2d, sqw_copy));
+        assertEqualToTol(obj.sqw_2d, sqw_copy);
+    end
+
+    function test_the_same_d2d_objects_are_equal(obj)
+        dnd_copy = obj.dnd_2d;
+        assertEqualToTol(obj.dnd_2d, dnd_copy);
+    end
+
+    function test_the_sqw_and_d2d_objects_are_not_equal(obj)
+        [ok, mess] = equal_to_tol(obj.dnd_2d, obj.sqw_2d);
+        assertFalse(ok);
+        assertEqual(mess, 'Objects being compared are not both sqw-type or both dnd-type');
+    end
+
+    function test_the_different_sqw_objects_are_not_equal(obj)
+        sqw_copy = obj.sqw_2d;
+        class_fields = properties(sqw_copy);
+        for idx = 1:numel(class_fields)
+            field_name = class_fields{idx};
+            
+            if isstruct(sqw_copy.(field_name))
+                sqw_copy.(field_name).test_field = 'test_value';
+            elseif isstring(sqw_copy.(field_name))
+                sqw_copy.(field_name) = 'test_value';
+            else
+                sqw_copy.(field_name) = [];
+            end
+            
+            [ok, mess] = equal_to_tol(obj.sqw_2d, sqw_copy);
+            assertFalse(ok, ['Expected ', field_name, ' to be not equal']);
+        end
+    end
+
+    function test_the_different_d2d_objects_are_not_equal(obj)
+        dnd_copy = obj.dnd_2d;
+        class_fields = properties(dnd_copy);
+        for idx = 1:numel(class_fields)
+            field_name = class_fields{idx};
+            if isstruct(dnd_copy.(field_name))
+                dnd_copy.(field_name).test_field = 'test_value';
+            elseif isstring(dnd_copy.(field_name))
+                dnd_copy.(field_name) = 'test_value';
+            else
+                dnd_copy.(field_name) = [];
+            end
+            
+            [ok, mess] = equal_to_tol(obj.dnd_2d, dnd_copy);
+            assertFalse(ok, ['Expected ', field_name, ' to be not equal']);
+        end
+    end
```

Comment:
Probably not a performance issue as it is only a small test, but you are retesting the entire object for inequality every time you change one field. Hence after the first changed field, the others may actually be equal but you will still get ok==false

Reply:
That's a good catch, the dnd copy should be inside the loop

Reply:
will fix SQW case too